src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -175,7 +182,36 @@ void MeshForwarder::HandleResolved(const Ip6::Address &aEid, Error aError)
     {
         mScheduleTransmissionTask.Post();
     }
+
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+exit:
+#endif
+    return;
+}
+
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+Error MeshForwarder::ForwardToBackboneLink(Message &aMessage, const Ip6::Address &aDst)
+{
+    Error    error = kErrorNone;
+    uint16_t dstRloc16;
+    uint8_t  ttl;
+
+    VerifyOrExit(Get<BackboneRouter::Local>().IsPrimary() && Get<BackboneRouter::Leader>().IsDomainUnicast(aDst),
+                 error = kErrorNoRoute);
+
+    IgnoreError(Get<AddressResolver>().Resolve(aDst, dstRloc16, /* aAllowAddressQuery */ false));
+    VerifyOrExit(dstRloc16 == Get<Mle::MleRouter>().GetRloc16(), error = kErrorNoRoute);
+
+    IgnoreError(aMessage.Read(Ip6::Header::kHopLimitFieldOffset, ttl));
+    ttl++;
+    aMessage.Write(Ip6::Header::kHopLimitFieldOffset, ttl);
+
+    IgnoreError(Get<Ip6::Ip6>().HandleDatagram(aMessage, nullptr, nullptr, false));
```

Comment:
```suggestion
    IgnoreError(Get<Ip6::Ip6>().HandleDatagram(aMessage, nullptr, nullptr, false));
```

The `Netif *aNetif` is current using `nullptr`, however this seems not totally correct:
- For messages sent by another device (e.g. MED child), `aNetif` should be set to `ThreadNetif`
- For messages sent by the PBBR itself, `aNetif` should be `nullptr`

The problem is that it's not easy to check if the message was initiated by PBBR itself or forwarded by PBBR at the current place because the information (receiving netif) was lost. 
Always using `nullptr` will bypass `mForwardingEnabled` checking, which maybe undesirable. 

Thoughts on how to address this issue? @librasungirl @abtink  @jwhui

Reply:
BTW, we are also violating `mForwardingEnabled` check in Ip6::SendRaw:

```
    error = HandleDatagram(aMessage, nullptr, nullptr, true);
```

Reply:
For checking whether or not it's from PBBR itself, may check whether or not the source of the message belongs to PBBR?

Here if the packet dest to DUA in another network, `forwardThread` should be false, there might be no chance to bypass `mForwardingEnabled` checking.
https://github.com/openthread/openthread/blob/ca450e541636692909183773beba8f55591a7185/src/core/net/ip6.cpp#L1256-L1262

In my understanding, `mForwardingEnabled` should be specific for forwarding rule in mesh and only router is allowed to forward packet it received via thread radio in mesh. I think it should be fine for SendRaw, from a different link into mesh.

Reply:
May check whether or not the source of the message belongs to PBBR to decide whether it's initiated by the PBBR itself or its MTD child?

Here if the packet destines to DUA in another network, `forwardThread` should be false, there might be no chance to bypass `mForwardingEnabled` checking.
https://github.com/openthread/openthread/blob/ca450e541636692909183773beba8f55591a7185/src/core/net/ip6.cpp#L1256-L1262

In my understanding, `mForwardingEnabled` should be specific for forwarding rule in mesh and only router is allowed to forward packet that it received via thread radio in mesh. I think it should be fine for `SendRaw`, from a different link into mesh.

Reply:
@librasungirl 

> In my understanding, mForwardingEnabled should be specific for forwarding rule in mesh and only router is allowed to forward packet it received via thread radio in mesh. I think it should be fine for SendRaw, from a different link into mesh.

+1 

> Here if the packet destines to DUA in another network, forwardThread should be false, there might be no chance to bypass mForwardingEnabled checking.

+1 

> May check whether or not the source of the message belongs to PBBR to decide whether it's initiated by the PBBR itself or its MTD child?

Setting `netif` seems useless according to your description above. So, maybe just fine to use `nullptr` for simplicity.

Reply:
no preference, either is fine to me.